src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -414,10 +418,20 @@ object CheckTypes extends Pass {
           errors append new InvalidConnect(info, mname, sx.loc.serialize, sx.expr.serialize)
         case sx: PartialConnect if !bulk_equals(sx.loc.tpe, sx.expr.tpe, Default, Default) =>
           errors append new InvalidConnect(info, mname, sx.loc.serialize, sx.expr.serialize)
-        case sx: DefRegister => sx.tpe match {
-          case AnalogType(w) => errors append new IllegalAnalogDeclaration(info, mname, sx.name)
-          case t if wt(sx.tpe) != wt(sx.init.tpe) => errors append new InvalidRegInit(info, mname)
-          case t =>
+        case sx: DefRegister => {
+          sx.tpe match {
+            case AnalogType(w) => errors append new IllegalAnalogDeclaration(info, mname, sx.name)
+            case t if wt(sx.tpe) != wt(sx.init.tpe) => errors append new InvalidRegInit(info, mname)
+            case t =>
+          }
+          sx.reset.tpe match {
+            case UIntType(IntWidth(w)) if (w == 1) =>
```

Comment:
Annoyingly, BigInt has no extractor/unapply methods.

If you think this type of extraction is sufficiently common, could also add an extractor in IntWidth for this itself. (Accessible with, like IntWidth.AsInt(1) and making sure to return None when can't return an Int.)

At the very least, should combine the cases via:
case UIntType(w) if w == IntWidth(1) || w == UnknownWidth